hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -1544,6 +1544,39 @@ public FSDataOutputStream append(Path f, int bufferSize) throws IOException {
   public abstract FSDataOutputStream append(Path f, int bufferSize,
       Progressable progress) throws IOException;
 
+  /**
+   * Append to an existing file (optional operation).
+   * @param f the existing file to be appended.
+   * @param appendToNewBlock whether to append data to a new block
+   * instead of the end of the last partial block
+   * @throws IOException IO failure
+   * @throws UnsupportedOperationException if the operation is unsupported
+   *         (default).
+   * @return output stream.
+   */
+  public FSDataOutputStream append(Path f, boolean appendToNewBlock) throws IOException {
+    return append(f, getConf().getInt(IO_FILE_BUFFER_SIZE_KEY,
+        IO_FILE_BUFFER_SIZE_DEFAULT), null, appendToNewBlock);
+  }
+
+  /**
+   * Append to an existing file (optional operation).
+   * This function is used for being overridden by some FileSystem like DistributedFileSystem
+   * @param f the existing file to be appended.
+   * @param bufferSize the size of the buffer to be used.
+   * @param progress for reporting progress if it is not null.
+   * @param appendToNewBlock whether to append data to a new block
+   * instead of the end of the last partial block
+   * @throws IOException IO failure
+   * @throws UnsupportedOperationException if the operation is unsupported
+   *         (default).
+   * @return output stream.
+   */
+  public FSDataOutputStream append(Path f, int bufferSize,
+      Progressable progress, boolean appendToNewBlock) throws IOException {
+    return append(f, bufferSize, progress);
```

Comment:
Thanks @M1eyu2018 for your report. I'm sorry that I didn't get your idea what you want to do, EC File supports `append` method? 

Maybe your can give a detailed explantation or share with us what problems you encountered.

Reply:
Thanks @ZanderXu for your report. My idea is that not only EC File, but also normalFile supports to append with new block while using appendToFile command.

Reply:
If the last block of the file is a partial block, can we append the file with `NEW_BLOCK` flag?

Reply:
Yes. For EC file, new block group will be created to append. For normal file, new block will be created to append.

Reply:
OMG, Can you add one UT with a partial last block to test `NEW_BLOCK` flag?

Reply:
It's in UT TestDFSShell#testAppendToFileWithOptionN.

Reply:
TestDFSShell#testAppendToFileWithOptionN didn't contains partial last block, because the block size is 1024 and the file size in the UT is `1024 * 1024`, so the last block is not partial. Maybe you can change the file size to `1024 * 1024 + 10` to verify it again.

Reply:
But I debug TestDFSShell#testAppendToFileWithOptionN. The block size is 134217728(128M) becuase DFS_BLOCK_SIZE_DEFAULT in HdfsConfiguration is 128*1024*1024.

Reply:
@ZanderXu Please help to review the code again, thank you very much!

Reply:
> TestDFSShell#testAppendToFileWithOptionN didn't contain a partial last block, because the block size is `1024` and the file size in the UT is `1024 * 1024`, so the last block is not partial block. Maybe you can change the file size to `1024 * 1024 + 10` to verify it again.

Hi @ZanderXu , there is a new Configuration. And block size is not `1024` but `128 * 1024 * 1024`. So the last block is partial.

Reply:
Thanks @tomscut for your correction. I will review this PR carefully later.

Reply:
Hi @ZanderXu , do you have any other suggestions? Thank you.